core/src/main/scala/com/banno/kafka/RecordStream.scala

```diff
@@ -595,6 +723,66 @@ object RecordStream {
         }
     }
 
+    private[RecordStream] type NeedsConsumer[F[_], A] =
+      Function[ConsumerApi[F, GenericRecord, GenericRecord], A]
+
+    private[RecordStream] case class StreamSelectorImpl[F[_]: Async, P[_[_], _], A](
+        whetherCommits: WhetherCommits[P],
+        topical: Topical[A, ?],
+    ) extends StreamSelector[F, NeedsConsumer[F, *], P, IncomingRecords[A]] {
+      override val present: NeedsConsumer[F, P[F, IncomingRecords[A]]] =
+        c => whetherCommits.extrude(recordStream(c, topical))
+
+      override val pastAndPresent: NeedsConsumer[F, PastAndPresent.Batched[F, P, A]] =
+        consumer =>
+          whetherCommits.pastAndPresent(
```

Comment:
If `G` was `Applicative`, could make this `final`, put it at the top, and express this relationship only one.